### PR TITLE
[wasm]  Add wasm specific assemblies to the bcl out directory

### PR DESCRIPTION
**Pull request by kjpou1**
- Small cleanup to the initialisation of the Bindings.

resolves https://github.com/mono/mono/issues/11856

### CODE REVIEW

File: sdks/wasm/Makefile [hunk 1]
```diff
@@ -110,20 +110,22 @@ main.exe: $(APP_SOURCES)
...
 
-binding_tests.dll: WebAssembly.Bindings.dll $(BINDING_TEST_SOURCES)
-	$(CSC) $(CSC_FLAGS) /unsafe -target:library -out:$@ /r:WebAssembly.Bindings.dll $(BCL_DEPS) /r:$(WASM_BCL_DIR)/nunitlite.dll $(BINDING_TEST_SOURCES)
+binding_tests.dll: .stamp-bindings $(BINDING_TEST_SOURCES)
+	$(CSC) $(CSC_FLAGS) /unsafe -target:library -out:$@ /r:$(WASM_BCL_DIR)/WebAssembly.Bindings.dll $(BCL_DEPS) /r:$(WASM_BCL_DIR)/nunitlite.dll $(BINDING_TEST_SOURCES)
 
-WebAssembly.Bindings.dll: bindings.cs
-	$(CSC) $(CSC_FLAGS) /target:library -out:$@ /r:$(WASM_BCL_DIR)/mscorlib.dll bindings.cs
+.stamp-bindings: bindings.cs
+	$(CSC) $(CSC_FLAGS) /target:library -out:$(WASM_BCL_DIR)/WebAssembly.Bindings.dll /r:$(WASM_BCL_DIR)/mscorlib.dll bindings.cs
```
Comment (vargaz) [inline sdks/wasm/Makefile]: Can't this simply be:
$(WASM_BCL_DIR)/WebAssembly.Bindings.dll: bindings.cs

File: sdks/wasm/Makefile [hunk 2]
```diff
@@ -253,12 +255,12 @@ package: build build-dbg-proxy build-framework-nuget
...
+	cp $(TOP)/sdks/out/wasm-bcl/wasm/WebAssembly.Bindings.dll tmp/framework
+	cp $(TOP)/sdks/out/wasm-bcl/wasm/WebAssembly.Bindings.pdb tmp/framework
 	cp WasmHttpMessageHandler.cs tmp
-	cp WebAssembly.Net.Http.dll tmp/framework
-	cp WebAssembly.Net.Http.pdb tmp/framework
-	cp WebAssembly.Framework.0.1.0-alpha.nupkg tmp/framework
+	cp $(TOP)/sdks/out/wasm-bcl/wasm/WebAssembly.Net.Http.dll tmp/framework
+	cp $(TOP)/sdks/out/wasm-bcl/wasm/WebAssembly.Net.Http.pdb tmp/framework
+	cp WebAssembly.Framework.0.1.0-alpha.nupkg tmp/framework	
 	cp sample.html tmp/
```
Comment (vargaz) [inline sdks/wasm/Makefile]: Extra whitespace.

File: sdks/wasm/Makefile [hunk 3]
```diff
@@ -284,3 +286,18 @@ canary:
 	/Applications/Google\ Chrome\ Canary.app/Contents/MacOS/Google\ Chrome\ Canary --remote-debugging-port=9222
 
 check-aot: do-aot-sample
+
+clean-mono:
+	rm -rf ./bin
```
Comment (vargaz) [inline sdks/wasm/Makefile]: This should just be clean:

File: sdks/wasm/Makefile [hunk 4]
```diff
@@ -284,3 +286,18 @@ canary:
...
+	rm -rf .stamp-build-debug-sample
+	rm -rf .stamp-build-test-suite
+	rm -rf .stamp-bindings
+	rm -rf .stamp-http
+	rm -rf WebAssembly.Bindings.dll
+	rm -rf WebAssembly.Net.Http.dll
+	rm -rf sample.dll
```
Comment (vargaz) [inline sdks/wasm/Makefile]: These don't exist anymore.

File: sdks/wasm/Makefile (hunk 3 above)
Comment (vargaz) [reply sdks/wasm/Makefile]: It can use $(RM) -r

File: sdks/wasm/packager.cs [hunk 5]
```diff
@@ -300,7 +300,7 @@ enum CopyType
 			Import (resolved, kind);
 		}
 		if (add_binding)
-			Import (ResolveFramework (BINDINGS_ASM_NAME + ".dll"), AssemblyKind.Framework);
+			Import (ResolveBcl (BINDINGS_ASM_NAME + ".dll"), AssemblyKind.Framework);
```
Comment (vargaz) [inline sdks/wasm/packager.cs]: Is the ResolveFramework code needed any more ?

File: sdks/wasm/Makefile (hunk 4 above)
Comment (kjpou1) [reply sdks/wasm/Makefile]: That should have never gone in.

File: sdks/wasm/Makefile (hunk 3 above)
Comment (kjpou1) [reply sdks/wasm/Makefile]: should have never gone in.

File: sdks/wasm/packager.cs (hunk 5 above)
Comment (kjpou1) [reply sdks/wasm/packager.cs]: Will remove it.

File: sdks/wasm/Makefile (hunk 1 above)
Comment (kjpou1) [reply sdks/wasm/Makefile]: fixed

File: sdks/wasm/Makefile (hunk 2 above)
Comment (kjpou1) [reply sdks/wasm/Makefile]: Fixed

File: sdks/wasm/packager.cs (hunk 5 above)
Comment (kjpou1) [reply sdks/wasm/packager.cs]: addressed in packager.cs

File: sdks/wasm/Makefile (hunk 4 above)
Comment (kjpou1) [reply sdks/wasm/Makefile]: Went ahead and left the clean in.

File: sdks/wasm/Makefile [hunk 6]
```diff
@@ -253,11 +251,11 @@ package: build build-dbg-proxy build-framework-nuget
 	rm tmp/release/.stamp-build
 	cp bindings.cs tmp
 	mkdir tmp/framework
-	cp WebAssembly.Bindings.dll tmp/framework
-	cp WebAssembly.Bindings.pdb tmp/framework
+	cp $(TOP)/sdks/out/wasm-bcl/wasm/WebAssembly.Bindings.dll tmp/framework
+	cp $(TOP)/sdks/out/wasm-bcl/wasm/WebAssembly.Bindings.pdb tmp/framework
 	cp WasmHttpMessageHandler.cs tmp
```
Comment (vargaz) [inline sdks/wasm/Makefile]: Do these still need to be copied, since they are in the bcl now ?

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build OS X i386

File: sdks/wasm/Makefile [hunk 7]
```diff
@@ -169,7 +169,7 @@ check-aot-mini run-aot-mini: build-aot-mini
 define AotTestTemplate
 
 build-aot-$(1): packager.exe runtime-tests.js main.exe WebAssembly.Bindings.dll binding_tests.dll $$(WASM_BCL_DIR)/nunitlite.dll $$(WASM_BCL_DIR)/tests/$(2)
-	mono packager.exe --emscripten-sdkdir=$$(EMSCRIPTEN_SDKDIR) --mono-sdkdir=$$(TOP)/sdks/out -appdir=bin/aot-$(1) --nobinding --builddir=obj/aot-$(1) --aot --template=runtime-tests.js main.exe WebAssembly.Bindings.dll binding_tests.dll $$(WASM_BCL_DIR)/tests/$(2)
```
Comment (vargaz) [inline sdks/wasm/Makefile]: The dependency needs to be updated too.

File: sdks/wasm/Makefile (hunk 6 above)
Comment (kjpou1) [reply sdks/wasm/Makefile]: Left those there so as not break existing projects that use the links.

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by kjpou1** [comment]
That goes for the WebAssembly.Net.Http.cs source as well or just the WebAssembly.Bindings.cs?

**Comment by luhenry** [comment]
@kjpou1 both would make sense, if you want to get them shipped as part of the archive.

**Comment by kjpou1** [comment]
Close this in favour of other solution as discussed with Ludovic.  The sources mentioned should not be part of the main branch.

**Pull request closed without merging**